src/main/resources/assets/carpet/scripts/distance_rc_1.sc

```diff
@@ -0,0 +1,55 @@
+import('math','_euclidean','_manhattan','_round');
+
+__config()->{
+    'commands'->{
+        'from <start_pos>'->'set_start',
+        'from <start_pos> to <end_pos>'->'_calculate',
+        'to <end_pos>'->_(end_pos)->(_calculate(null,end_pos))
+    }
+};
+
+global_positions={};
+
+set_start(pos)->(
+    global_positions:player() = pos;
+    print(player(),format('gi Initial point set to: ', 'g '+pos,'?/tp '+(str(pos)-','-'['-']')))
+);
+
+_calculate(start_pos, end_pos)->(
+    if(start_pos!=null,
+        set_start(start_pos);
+        calculate(end_pos),
+        if(has(global_positions,player()),
+            calculate(end_pos),
+            print(player(),format('r There was no initial point for '+player()));
+            set_start(end_pos)
+        )
+    )
+);
+
+__on_player_places_block(player, item_tuple, hand, block) ->(
+    if(block=='brown_carpet' && system_info('world_carpet_rules'):'carpets'=='true',
+        if(player~'sneaking',
```

Comment:
If the player isn't sneaking, it should calculate with previous start pos (or set start pos if not present), right?

Reply:
ye, thats what it does mate.

Reply:
Oops, read it wrong. Now after actually testing: Maybe not send an error if no initial pos was available on first carpet.

Reply:
It send error message and registers that as first pos.